nova_cli/src/main.rs

```diff
@@ -55,55 +58,56 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         Command::Eval { verbose, paths } => {
             let allocator = Default::default();
 
-            let mut agent = Agent::new(
+            let mut agent = BoxedAgent::new(
                 Options {
                     disable_gc: false,
                     print_internals: verbose,
                 },
                 &DefaultHostHooks,
             );
-            {
+            assert!(!paths.is_empty());
+            agent.with(|agent, root_realms| {
                 let create_global_object: Option<fn(&mut Realm) -> Object> = None;
                 let create_global_this_value: Option<fn(&mut Realm) -> Object> = None;
                 initialize_host_defined_realm(
-                    &mut agent,
+                    agent,
                     create_global_object,
                     create_global_this_value,
                     Some(initialize_global_object),
                 );
-            }
-            let realm = agent.current_realm_id();
-
-            // `final_result` will always be overwritten in the paths loop, but
-            // we populate it with a dummy value here so rustc won't complain.
-            let mut final_result = Ok(Value::Undefined);
-
-            assert!(!paths.is_empty());
+                let realm = agent.current_realm_id();
+                root_realms.push(realm);
+            });
+            agent.gc();
```

Comment:
**note:** I added GC calls here and there to essentially test the GC. We probably don't want to leave these changes in as they are.

Reply:
I shuffled these around a bit to make them less impactful, and then disabled GC in test262_runner; the test run takes ~15 seconds on my computer but ~40s if GC is run.